### ISSUE TITLE
SIGSEGV On Linux (Manjaro - Archlinux)

### ISSUE THREAD

**Issue by lmichaudel** [SoftwareGuy/ENet-CSharp#24]
Hi, I am trying to use this ENet implementation into the Godot Engine, however I get a crash on linux system (manjaro - archlinux), work well on the windows build.
The same crash also happens using the "classic" ENet-CSharp package.
Here is the crash :
```
=================================================================
	Native Crash Reporting
=================================================================
Got a SIGSEGV while executing native code. This usually indicates
a fatal error in the mono runtime or one of the native libraries 
used by your application.
=================================================================

=================================================================
	Native stacktrace:
=================================================================
	0xb8e86b - ./p2p.x86_64 : 
	0xb8ebf9 - ./p2p.x86_64 : 
	0xb818bf - ./p2p.x86_64 : 
	0xaf9069 - ./p2p.x86_64 : 
	0x24c6b60 - ./p2p.x86_64 : enet_peer_reset_queues
	0x7f526497033a - /media/linux/libenet.so : enet_host_create
	0x41961a98 - Unknown

=================================================================
	Telemetry Dumper:
=================================================================
Pkilling 0x139991845959232x from 0x139992361540992x
Could not exec mono-hang-watchdog, expected on path '/media/linux/data_enet-p2p/Mono/etc/../bin/mono-hang-watchdog' (errno 2)
Entering thread summarizer pause from 0x139992361540992x
Finished thread summarizer pause from 0x139992361540992x.
Failed to create breadcrumb file (null)/crash_hash_0x3ce389cc2

Waiting for dumping threads to resume
```
I use the libenet.so from the release, and ENetCsharp.dll has been compiled by me targetting .NET 4.7.2

**Comment by Vincenz099** [comment]
Need your code, where does it exactly crash, during your host creation call i suppose? 
What is your setup? settings? etcetc. this is stabbing in the dark right now.

**Comment by lmichaudel** [comment]
The code is quite simple, happens when these lines (mostly the server.Create()) are called, which happens in the _Ready functions of Godot, basically called on the first frame of the game

```cs
Address address = new Address();
address.Port = PORT;

Host server = new Host();
server.Create(address, MAX_CLIENT);
```

My Godot version is the Mono one, which supports C#. I export the game (= build it) inside the default Linux (X11 to be more precise) settings. This give me a .x86_64 file. At the root of the builded project, I put the libenet.so and the ENetCSharp.dll. I also have mono crashlog.

My Linux is the latest version of Manjaro, which is based on Archlinux. Also it run on virtualbox, but I also tested running the game on a real manjaro install and the same happens.

Let me know if you need more informations
(the .json crashlog is inside a zip, as I can't attach json files)
[mono_crash.3ce389b8a.5.zip](https://github.com/SoftwareGuy/ENet-CSharp/files/6561430/mono_crash.3ce389b8a.5.zip)

**Comment by lmichaudel** [comment]
Also here is the enet-debug.log. The crash happens when clearing queues : 
```
2021-05-28 15:38:08 [TRACE] [enet_initialize:4468] Unix-like Environment: Initialization
2021-05-28 15:38:08 [TRACE] [enet_host_create:3902] Creating ENet host object.
2021-05-28 15:38:08 [TRACE] [enet_host_create:3932] Socket creation: IPv4 & IPv6 dual stack mode
2021-05-28 15:38:08 [TRACE] [enet_host_create:3951] Buffers: Setting up
2021-05-28 15:38:08 [TRACE] [enet_host_create:3957] Socket: Setting up options
2021-05-28 15:38:08 [TRACE] [enet_host_create:3963] Address: Setting up
2021-05-28 15:38:08 [TRACE] [enet_host_create:3967] Channels: Setting up
2021-05-28 15:38:08 [TRACE] [enet_host_create:4002] Startup: Clearing queues
```

**Comment by SoftwareGuy** [comment]
Alright, so from what you've told us this crash is happening on a Manjaro installation. I do have an Arch Linux headless machine here in my local network that I could try a ENet C# basic application on, ie. one that spawns a dummy server just for testing purposes. That could help us isolate if it's something in Godot, or if the native library is busted. Can you check on your end if the sample code included in the repository triggers this crash too, or it's solely Godot-related?

ENet Native blobs are compiled on a GitHub Ubuntu instance (18.04 LTS I believe? Or was it 20.04 LTS...) but I'm pretty sure that Manjaro doesn't have any crazy differences with the native C (libc) library. If there was something messed up then other stuff would be breaking too.

I've been using native ENet inside Unity on Linux runtime for some time now with this wrapper and I haven't been experiencing this SIGSEGV. And that's been on Arch, Debian, Ubuntu of different versions.

**Comment by lmichaudel** [comment]
I have made a bunch of test. First I made a basic console c# app, and the same app in godot to compare them.
The Basic C# App was compiled on the Linux computer, and ran perfectly without any errors
However the Godot app still had the same crash, so this is definitely caused by Godot.
I will try to compile the Godot game for Linux ON the linux computer, as it was done on Windows.
I will also ask about this on the Godot discord.

TLDR; It comes from Godot
